pytorch_lightning/utilities/xla_device_utils.py

```diff
@@ -0,0 +1,57 @@
+import functools
+import importlib
+from multiprocessing import Process, Queue
+
+TORCHXLA_AVAILABLE = importlib.util.find_spec("torch_xla") is not None
+if TORCHXLA_AVAILABLE:
+    import torch_xla.core.xla_model as xm
+else:
+    xm = None
+
+
+def inner_f(queue, func, **kwargs):
+    try:
+        queue.put(func(**kwargs))
+    except Exception as _e:
+        import traceback
+        traceback.print_exc()
+        queue.put(None)
+
+
+def pl_multi_process(func):
```

Comment:
there is a similar, if not identical function in `tests/base/develop_utils.py`
do we need both? I cannot see an obvious difference besides the inner_f being defined outside.

Reply:
I basically used the other function itself as reference. the one in `develop_utils` is just for tests right?

Reply:
yeah, but couldn't you delete the one from tests and then import from here?

Reply:
The `pl_multi_process_test` varies a bit compared to this function. It has an `assert` statement within and returns either `1` or `-1` for the test. This one is meant to return the device type on `None`.